### PR TITLE
ad9361: switch to common drivers

### PR DESCRIPTION
**Pull request by cristianpop-analog**
Switch to axi_adc_core, axi_dac_core, axi_dmac interface.
Use common drivers entirely and remove duplicated implementation, for AD9361.

Signed-off-by: Cristian Pop <cristian.pop@analog.com>

### CODE REVIEW

**Comment by cristianpop-analog** [comment]
rebase master-adrv9009_dac_dma_example

File: projects/ad9361/src/ad9361_util.h [hunk 1]
```diff
@@ -55,15 +55,13 @@
...
 #define SUCCESS									0
-#define ARRAY_SIZE(arr)							(sizeof(arr) / sizeof((arr)[0]))
 #define min(x, y)								(((x) < (y)) ? (x) : (y))
 #define min_t(type, x, y)						(type)min((type)(x), (type)(y))
 #define max(x, y)								(((x) > (y)) ? (x) : (y))
 #define max_t(type, x, y)						(type)max((type)(x), (type)(y))
 #define clamp(val, min_val, max_val)			(max(min((val), (max_val)), (min_val)))
 #define clamp_t(type, val, min_val, max_val)	(type)clamp((type)(val), (type)(min_val), (type)(max_val))
 #define DIV_ROUND_UP(x, y)						(((x) + (y) - 1) / (y))
-#define DIV_ROUND_CLOSEST(x, divisor)			(((x) + (divisor) / 2) / (divisor))
```
Comment (stefpopa) [inline projects/ad9361/src/ad9361_util.h]: Curious: is there any reason why these macros were removed?

Comment (cristianpop-analog) [reply projects/ad9361/src/ad9361_util.h]: They are already defined in drivers/util/util.h

Comment (stefpopa) [reply projects/ad9361/src/ad9361_util.h]: Ok, thanks for the clarification!

**Comment by dbogdan** [comment]
In the list of commits, I can see Stefan's ADRV9009 related commits too - they were already merged to master, so we should do a rebase one more time.

**Comment by dbogdan** [comment]
v3:
   - Apply the commits on top of the current master branch.

**Pull request merged**